src/custom-elements/src/components/label/label.tsx

```diff
@@ -38,7 +49,37 @@ export class Label {
     addMutationObserver.call(this);
   }
 
+  componentWillLoad() {
+    this.helpButtonId = `help-button-${uuid4()}`;
+    this.helpPopoverId = `help-popover-${uuid4()}`;
+  }
+
+  componentDidLoad(): void {
+    this.el
+      .querySelector(`#${this.helpButtonId}`)
+      .addEventListener('click', () => (document.querySelector(`#${this.helpPopoverId}`) as any).toggle());
+  }
+
   render() {
+    const infoIcon = this.infoIcon ? (
+      <div class="chi-label__help">
+        <chi-button id={this.helpButtonId} type="icon" size="xs" variant="flat" alternative-text="Help">
+          <chi-icon icon="circle-info-outline"></chi-icon>
+        </chi-button>
+        <chi-popover
+          id={this.helpPopoverId}
+          position="top"
+          variant="text"
+          arrow
+          reference={`#${this.helpButtonId}`}
+        >
+          {this.infoIconMessage}
+        </chi-popover>
+      </div>
+    ) : (
+      ''
+    );
```

Comment:
I would move this to a function `this._getInfoIcon`, we can avoid complexity in render function that in the future is going to make SonarQube complain about high complexity functions.
```suggestion
    _getInfoIcon() {
      return this.infoIcon ? (
      <div class="chi-label__help">
        <chi-button id={this.helpButtonId} type="icon" size="xs" variant="flat" alternative-text="Help">
          <chi-icon icon="circle-info-outline"></chi-icon>
        </chi-button>
        <chi-popover
          id={this.helpPopoverId}
          position="top"
          variant="text"
          arrow
          reference={`#${this.helpButtonId}`}
        >
          {this.infoIconMessage}
        </chi-popover>
      </div>
    ) : (
      ''
    );
    }
    
    
    ....
    
    
    const infoIcon = this._getInfoIcon();
```